extension/docs/CancelRefundPayment.md

```diff
@@ -0,0 +1,59 @@
+**Please see [Integration Guide](IntegrationGuide.md) first before continuing with this document.**
+
+## Cancel or refund a payment
+
+1. Shop updates a payment with [a new transaction](https://docs.commercetools.com/http-api-projects-payments#add-transaction)
+    1. for refund
+        ```
+        {
+          "version": ${currentVersion},
+          "actions": [
+            {
+              "action": "addTransaction",
+              "transaction": {
+                "type": "Refund",
+                "amount": {
+                  "currencyCode": "§{currencyCode}",
+                  "centAmount": ${centAmount}
+                },
+                "state": "Initial"
+              }
+            }
+          ]
+        }
+        ```
+    1. for cancel authorization
+        ```
+        {
+          "version": ${currentVersion},
+          "actions": [
+            {
+              "action": "addTransaction",
+              "transaction": {
+                "type": "CancelAuthorization",
+                "amount": {
+                  "currencyCode": "§{currencyCode}",
+                  "centAmount": ${centAmount}
+                },
+                "state": "Initial"
+              }
+            }
+          ]
+        }
+        ```
+    Both transaction types works the same, because Adyen will always pick the right action for the current transaction.
+    If the transaction is not authorized yet, Adyen will try to cancel.
+    If the transaction is already charged, Adyen will do refund.
+1. Extension module finds the first `Payment.transactions` with a transaction `type='Charge' and state='Success'`. 
+Extension module takes `amount` and `transactionId` from this transaction and makes a Cancel or refund request.  
+1. Extension module saves following information to the payment object:
+    * `Payment.interfaceInteractions.type='cancelOrRefund'` contains request and response with Adyen
```

Comment:
So it saves the type which is `cancelOrRefund `, request to Adyen, and response from Adyen?

Reply:
Yes, we always did this.

Reply:
https://github.com/commercetools/commercetools-adyen-integration/pull/58/commits/6a9e46f76ebc8e3e514e60c8326d6b8716703ad5#diff-bfa0dfd06e6d73d9f7a2d7f9d0a6b080R50